### ISSUE TITLE
Using /etc/mysql as a volume does not work

### ISSUE THREAD

**Issue by RulerOf** [ampache/ampache-docker#24]
Every time I try to use a volume for the `/etc/mysql` directory in the container, the application fails to start because the database never comes online:

```
[root@drew-metal content-engine]# docker logs -f ampache
=> An empty or uninitialized MySQL volume is detected in /var/lib/mysql
=> Installing MySQL ...
=> Done!
=> Waiting for confirmation of MySQL service startup
=> Waiting for confirmation of MySQL service startup
=> Waiting for confirmation of MySQL service startup
=> Waiting for confirmation of MySQL service startup
=> Waiting for confirmation of MySQL service startup
=> Waiting for confirmation of MySQL service startup
```

It appears to be that the files needed are already in the image when it starts:

```
[root@drew-metal content-engine]# docker create ampache/ampache:latest
a42e08837dc3f229a92c5ad8555c658f806df46924809f4203c45440695873cd
[root@drew-metal content-engine]# docker export a42e08837dc3f229a92c5ad8555c658f806df46924809f4203c45440695873cd | tar t | grep etc/mysql
etc/mysql/
etc/mysql/conf.d/
etc/mysql/conf.d/.keepme
etc/mysql/conf.d/mysqld_safe_syslog.cnf
etc/mysql/debian-start
etc/mysql/debian.cnf
etc/mysql/my.cnf
[root@drew-metal content-engine]# docker rm a42e08837dc3f229a92c5ad8555c658f806df46924809f4203c45440695873cd
a42e08837dc3f229a92c5ad8555c658f806df46924809f4203c45440695873cd
```

So they go away when I attach a volume to that path, and the database doesn't work.

**Comment by RulerOf** [comment]
I've been using volumes incorrectly because I made some assumptions based on how other popular docker images are built.

If the source of a volume doesn't yet exist, [the contents of the destination inside the container image are copied to it](https://docs.docker.com/engine/reference/builder/#volume) when first creating the volume:

>The docker run command initializes the newly created volume with any data that exists at the specified location within the base image.

I created empty directories for my volumes before creating the container which short-circuited the file copy behavior.

**Comment by 75th** [comment]
Sorry to necro this thread, but I have this exact issue with a twist. I'm trying to use the docker-compose file in Container Manager in Synology DSM. If I create the data folders ahead of time, I get the same "Waiting for confirmation of MySQL service startup" loop. But if I *don't* create them, I get this error:

```Bind mount failed: '/volume1/Ampache/data/mysql' does not exists```

([*sic*] on "exists")

I went looking to learn more about Docker stuff to understand this issue, and I immediately found [this](https://docs.docker.com/storage/volumes/):

> Volumes have several advantages over bind mounts:

So the docs say volumes are an alternative to bind mounts, but I'm getting bind mount errors when trying to use volumes. At this point I threw up my hands and came here. By all means tell me if I should go bug the Synology forum, but the Synology forum is pretty useless in my experience, and I thought maybe there's a chance this is still Ampache-specific.

Here's the docker-compose YAML I'm using. Changed a port and omitted the last line, which DSM didn't like.

<details>
<summary>My docker-compose</summary>

```
version: '3'

services:
  ampache:
    image: ampache/ampache:latest
    container_name: ampache
    restart: unless-stopped
    ports:
      - 50180:80
    volumes:
      - ./data/config:/var/www/config
      - ./data/log:/var/log/ampache
      - ./data/media:/media
      - ./data/mysql:/var/lib/mysql
```
</details>
